packages/dev/core/src/Meshes/GaussianSplatting/gaussianSplattingMesh.ts

```diff
@@ -74,16 +82,45 @@ export class GaussianSplattingMesh extends Mesh {
      * @returns the current mesh
      */
     public render(subMesh: SubMesh, enableAlphaMode: boolean, effectiveMeshReplacement?: AbstractMesh): Mesh {
+        if (!this.material) {
+            this._material = new GaussianSplattingMaterial(name + "_material", this._scene);
+            this.material = this._material;
+        }
+
         const frameId = this.getScene().getFrameId();
-        if (frameId !== this._frameIdLastUpdate && this._worker && this._scene.activeCamera) {
-            this._frameIdLastUpdate = frameId;
+        if (frameId !== this._frameIdLastUpdate && this._worker && this._scene.activeCamera && this._canPostToWorker) {
             this.getWorldMatrix().multiplyToRef(this._scene.activeCamera.getViewMatrix(), this._modelViewMatrix);
-            this._worker.postMessage({ view: this._modelViewMatrix.m });
+
+            const dot = this._lastProj[2] * this._modelViewMatrix.m[2] + this._lastProj[6] * this._modelViewMatrix.m[6] + this._lastProj[10] * this._modelViewMatrix.m[10];
+            if (Math.abs(dot - 1) >= 0.01) {
+                this._frameIdLastUpdate = frameId;
+                this._canPostToWorker = false;
+                this._lastProj = this._modelViewMatrix.m.slice(0);
+                this._worker.postMessage({ view: this._modelViewMatrix.m, depthMix: this._depthMix }, [this._depthMix.buffer]);
+            }
         }
 
         return super.render(subMesh, enableAlphaMode, effectiveMeshReplacement);
     }
 
+    /**
+     * @internal
+     */
+    public _bind(subMesh: SubMesh, effect: Effect, fillMode: number, allowInstancedRendering = true): Mesh {
+        if (this._covariancesATexture) {

```

Comment:
last question about this part :-)

I wonder if the material should pull the data instead in its bind function ? It makes creating other GSPlat mat or shader mat equivalent simpler I guess ?